### PR TITLE
Add GridType to SIR and to IIR

### PR DESCRIPTION
**Pull request by Stagno**
## Technical Description

- Add grid type (enum) to SIR and IIR classes. Need to construct them with a grid type.
- Update also protobuf, (de)serialization, tests, python and python tests.

### Resolves / Enhances

First bullet of #422

### CODE REVIEW

File: dawn/src/dawn/IIR/StencilInstantiation.cpp [hunk 1]
```diff
@@ -48,22 +48,24 @@ namespace iir {
 //===------------------------------------------------------------------------------------------===//
 
 StencilInstantiation::StencilInstantiation(
-    sir::GlobalVariableMap const& globalVariables,
+    const ast::GridType gridType, sir::GlobalVariableMap const& globalVariables,
```
Comment (lukasm91) [inline dawn/src/dawn/IIR/StencilInstantiation.cpp]: as we decided not to decide on where to put the const, I would just adapt to the surrounding code. Here: put it to the right.

File: dawn/examples/python/tridiagonal_solve.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ def create_vertical_region_stmt3() -> VerticalRegionDeclStmt:
 
 (options, args) = parser.parse_args()
 
-hir = make_sir("tridiagonal_solve.cpp", [
+hir = make_sir(GridType.Structured, "tridiagonal_solve.cpp", [
```
[the remark this answers is not in the document]
Comment (Stagno) [reply dawn/examples/python/tridiagonal_solve.py]: I think the sir printer only prints AST

File: dawn/src/dawn/IIR/StencilInstantiation.cpp (hunk 1 above)
Comment (Stagno) [reply dawn/src/dawn/IIR/StencilInstantiation.cpp]: done, thx for the hint

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Comment by Stagno** [comment]
launch jenkins

**Pull request merged**